### ISSUE TITLE
Impact Particles from gunshots stop appearing 

### ISSUE THREAD

**Issue by Kitakazi** [TheGameCreators/GameGuruRepo#4865]
This issue has been present for about a month or so but it's hard to reproduce immediately. This is because if you open Max and load a level bullet impacts will function as normal. However after an extended period of time they will stop appearing all together. The bullet holes will still appear but all impacts will be gone, including blood. They will 100% of the time disappear, it just takes a few minutes. 

The best way I've found to reproduce this is too load a level then press the "Back to storyboard" button let it load my project then go back to the level and test game. Usually the impact particles will stop appearing at that point. 

I have also seen where bullet impacts appear at the start of the level but after a while, during the same Test Game, they will stop appearing altogether.

**Comment by Kitakazi** [comment]
Havent noticed this since last update seems to be fixed now.

**Comment by Kitakazi** [comment]
its happening again.

**Comment by LeeBamberTGC** [comment]
@Kitakazi This may relate to muzzle flashes from enemies also failing to make an appearance sometimes.  This is likely to be fixed when we drop decal particles in favor of real ones.

**Comment by Kitakazi** [comment]
Maybe related to that, but I have a feeling something strange is going on with gunshots. Sometimes when the impact decals stop appearing, and I shoot at an object near a floor or a floor entity itself, It will still spawn the bullet hole but I can sometimes hear the bullet impacting water like its going through the entity, and impacting water far below the entity/terrain. I wonder if the impact decals are appearing at that location (wherever its hitting water) instead? I dunno maybe not related this issue at all just something I noticed.

**Comment by Kitakazi** [comment]
@LeeBamberTGC Ok I think I figured out how to reproduce this one and what the cause is. 

If the player walks into water then bullet impacts will stop appearing. After the player walks through water bullet impacts will slowly stop appearing. Once this happens, even reloading the level will not make the bullet impacts come back. Max has to be restarted completely. I know you said it will probably be fixed when the particle impacts come but yea just wanted to let you know. Here is an FPM with stock entities so you can reproduce. 
[test 1.zip](https://github.com/TheGameCreators/GameGuruRepo/files/12511205/test.1.zip)
video of it happening - https://youtu.be/LOZpsypEv-8


Just run into the water and run back, you will see that impact particles disappear. If you go back into the water for a 2nd or 3rd time the impact particles begin to appear less and less until the never appear again.

**Comment by LeeBamberTGC** [comment]
@Kitakazi Thanks for the test level, am able to reproduce every time, perfect, thanks!  Will post again when fixed.

**Comment by LeeBamberTGC** [comment]
@Kitakazi Was a beastie this one, but fixed it. Was caused by water ripples setting cull mode on them, but when decal reused for impact decals, the cull mode was not properly reset and thus not visible. Thanks for the repro!!  Fixed for next EXP build.  As a bonus, I also fixed it so when bullets hit the bullethole scorch decal, it does not invalidate the material index, thus allowing the particle decal effect to show even if shooting into the same spot on a solid surface, good feedback to know where the bullet struck:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/f5c29712-d902-43b9-b1bd-f7ea0e8907f4)
